### PR TITLE
re-ordered tutorials on webpage

### PR DESCRIPTION
**Pull request by mmahsereci**
# In a Nutshell
tutorial order now similar to readme (solvers first, then lower level structure). No changes in the actual tutorials yet.

# Detailed Description

# Related Issues
Closes #...

### CODE REVIEW

**Comment by mmahsereci** [comment]
@JonathanWenger could you perhaps especially check the text below Linear solvers. In particular the last sentence as I am not entirely sure which solvers we provided atm. Thanks

File: docs/source/tutorials.rst [hunk 1]
```diff
@@ -20,43 +20,33 @@ Getting Started
...
-
-
-
-Linear Algebra
+Linear Solvers
 --------------
 
-Linear algebra is fundamental to arguably most of scientific computation. ProbNum defines (finite-dimensional) linear
-operators in a memory-efficient manner and implements probabilistic numerical methods to solve linear systems.
+Solving linear systems is arguable one of the most fundamental computations in statistics and machine learning
```
Comment (JonathanWenger) [inline docs/source/tutorials.rst:26]:
```suggestion
Solving linear systems is arguably one of the most fundamental computations in statistics, machine learning and numerics. For example, linear systems arise when inferring parameters in statistical models or during model training.
ProbNum provides a family of linear solvers, which infer either the inverse system matrix or the solution directly, while quantifying their uncertainty.
```

File: docs/source/tutorials.rst [hunk 2]
```diff
@@ -78,6 +68,29 @@ Gaussian filtering and smoothing in more complex dynamical systems.
    tutorials/filtsmooth/particle_filtering
 
 
+Linear Operators
+----------------
+
+Linear algebra is fundamental to arguably most of scientific computation. ProbNum defines (finite-dimensional) linear
```
Comment (JonathanWenger) [inline docs/source/tutorials.rst:74]:
```suggestion
Linear algebra is fundamental essentially all of scientific computation. ProbNum implements (finite-dimensional) linear operators in a memory-efficient manner with support for lazy arithmetic.
```

**Review by JonathanWenger — CHANGES_REQUESTED** [review]

File: docs/source/tutorials.rst (hunk 2 above)
Comment (mmahsereci) [reply docs/source/tutorials.rst:75]: That first part of the sentence has a word missing I believe, and "fundamental" and "essentially" sound a bit redundant to me. What about 

"Linear algebra is fundamental to virtually all of scientific computation." 

or 

"Linear algebra is essentially to virtually all of scientific computation."

I'll change the second sentence as proposed.

Comment (mmahsereci) [reply docs/source/tutorials.rst]: I used the second version.

**Review by JonathanWenger — APPROVED** [review]

**Pull request merged**